### PR TITLE
Defect/547 sandbox session continuation

### PR DESCRIPTION
**Pull request by esotrope**

### CODE REVIEW

File: webapp/src/main/java/org/opentestsystem/rdw/reporting/security/sandbox/SandboxUserDetailsService.java [hunk 1]
```diff
@@ -48,9 +48,14 @@ public UserDetails loadUserDetails(final SandboxAuthenticationToken token) throw
         final Tenant sandbox = sandboxRepository.findByKey(token.getSandboxKey());
         TenantContextHolder.setTenantId(sandbox.getId());
 
-        // TODO allow way to log back in again with same key through extra param?
-        final String usernameAndId = UUID.randomUUID().toString();
```
Comment (esotrope) [inline webapp/src/main/java/org/opentestsystem/rdw/reporting/security/sandbox/SandboxUserDetailsService.java]: as i guessed this was just overwriting the requested uuid

File: webapp/src/main/java/org/opentestsystem/rdw/reporting/security/sandbox/SandboxUserDetailsService.java [hunk 2]
```diff
@@ -48,9 +48,14 @@ public UserDetails loadUserDetails(final SandboxAuthenticationToken token) throw
...
-
+        // User ID is a combination of the requested role and UUID
+        final String usernameAndId = String.format(
+                "%s,%s",
+                token.getRole(),
```
Comment (esotrope) [inline webapp/src/main/java/org/opentestsystem/rdw/reporting/security/sandbox/SandboxUserDetailsService.java]: went ahead and made the user id a composite of the role and uuid. that make sense to everyone else? @ftdholbrook @malaffoon

File: webapp/src/main/resources/application.sql.yml [hunk 3]
```diff
@@ -41,23 +41,19 @@ sql:
...
-          g.sequence as grade_order
+            'SchoolAdministrator' as type,
+            2 as type_order,
+            CONCAT('school_', natural_id) AS id,
+            'School' as organization_type,
+            s.natural_id as organization_natural_id,
+            s.id as organization_id,
+            s.name as organization_name,
+            g.code as grade_code,
+            MAX(g.sequence) as grade_order
```
Comment (esotrope) [inline webapp/src/main/resources/application.sql.yml]: thanks @malaffoon you are an sql wizard

File: CHANGELOG.md [hunk 4]
```diff
@@ -6,6 +6,9 @@
     * Changes the configuration properties ... a lot.
 * Ingest pipeline administration.
 * Remove stale report cleanup task.
+* Changes student field and filter related text configuration locations in en.json:
+    * aggregate-report-form.field.* -> common.student-field.*
+    * common.filters.student.* -> common.student-field.*
```
Comment (malaffoon) [inline CHANGELOG.md]: 👍

File: webapp/src/main/java/org/opentestsystem/rdw/reporting/security/sandbox/SandboxUserDetailsService.java (hunk 2 above)
Comment (malaffoon) [reply webapp/src/main/java/org/opentestsystem/rdw/reporting/security/sandbox/SandboxUserDetailsService.java]: i guess i don't care but wasn't UUID sufficient?

Comment (esotrope) [reply webapp/src/main/java/org/opentestsystem/rdw/reporting/security/sandbox/SandboxUserDetailsService.java]: just felt weird to technically allow a user to log in with the same uuid and a different role and share the data between those "accounts"

File: webapp/src/main/webapp/src/app/app.component.ts [hunk 5]
```diff
@@ -61,13 +61,13 @@ export class AppComponent implements OnInit {
...
-                this.router.navigateByUrl('/error');
-                return _throw(error);
-              }),
               tap(settings => {
                 this.languageStore.configuredLanguages = settings.uiLanguages;
                 this.initializeAnalytics(settings.analyticsTrackingId);
+              }),
+              catchError(error => {
+                this.router.navigateByUrl('/error');
+                return _throw(error);
```
Comment (malaffoon) [inline webapp/src/main/webapp/src/app/app.component.ts]: Just for my own edification: the order doesn't technically matter right? Just looks nicer and follow convention?

Comment (esotrope) [reply webapp/src/main/webapp/src/app/app.component.ts]: correct

**Pull request merged**